### PR TITLE
ENH: stats.describe: add array API support

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Towards https://github.com/scipy/scipy/issues/20544

#### What does this implement/fix?
Adds array API support to `scipy.stats.describe`.

#### Additional information
I think our policy of translating existing tests rather than rewriting is good, but it was pretty cumbersome for this one. Most of what we really need is one big property-based test showing that the results of `describe` match the results of the functions it calls. Those functions are tested elsewhere.

### CODE REVIEW

**Comment by mdhaber** [comment]
CI failure appears to be unrelated.

File: scipy/stats/tests/test_stats.py [hunk 1]
```diff
@@ -5970,40 +5970,48 @@ def test_ttest_1samp_popmean_array(xp):
...
+        xp_assert_close(sk, skc)
+        xp_assert_close(kurt, kurtc)
+
+        # n, mm, m, v, sk, kurt = stats.describe(x.T, axis=1)
+        # assert_equal(n, nc)
+        # assert_equal(mm, mmc)
+        # assert_equal(m, mc)
+        # assert_equal(v, vc)
+        # assert_array_almost_equal(sk, skc, decimal=13)
+        # assert_array_almost_equal(kurt, kurtc, decimal=13)
```
Comment (j-bowhay) [inline scipy/stats/tests/test_stats.py:6012]: Is this meant to be commented out?

Comment (mdhaber) [reply scipy/stats/tests/test_stats.py:6012]: Oops! Nope, I often comment out, run tests, uncomment, run tests, etc... Forgot to do this last bit. Anything else to include in the next commit?

Comment (j-bowhay) [reply scipy/stats/tests/test_stats.py:6012]: I think the rest looked reasonable on a first pass

Comment (mdhaber) [reply scipy/stats/tests/test_stats.py]: Great. Converted these, and it looks like tests are passing.

**Comment by j-bowhay** [comment]
Thanks @mdhaber planning to review this and other from Wednesday onwards

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -1402,20 +1402,27 @@ def describe(a, axis=0, ddof=1, bias=True, nan_policy='propagate'):
...
     if contains_nan and nan_policy == 'omit':
-        a = ma.masked_invalid(a)
-        return mstats_basic.describe(a, axis, ddof, bias)
+        if is_numpy(xp):
+            a = ma.masked_invalid(a)
+            return mstats_basic.describe(a, axis, ddof, bias)
+        else:
+            msg = ("Use of `nan_policy` and `keepdims` "
+                   "is incompatible with non-NumPy arrays.")
+            raise NotImplementedError(msg)
```
Comment (j-bowhay) [inline scipy/stats/_stats_py.py:1417]: Can you add test coverage for this branch, checking locally it doesn't appear that this is hit

**Review by j-bowhay — COMMENTED** [review]
One minor comment then this is good go

File: scipy/stats/_stats_py.py (hunk 2 above)
Comment (mdhaber) [reply scipy/stats/_stats_py.py]: Oops, turns out this is covered by `_contains_nan` and doesn't need to appear in functions individually. 

While testing locally, I see that gh-20467 caused a failure and worked around it with `_array_api.size`.

**Review by j-bowhay — APPROVED** [review]
Thanks @mdhaber

**Pull request merged**